influence_benchmark/stats/preferences_per_iteration.py

```diff
@@ -117,40 +125,42 @@ def analyze_run(
     data_path = PROJECT_DATA / "trajectories" / run_name
     iterations = sorted(int(d.name) for d in data_path.iterdir() if d.is_dir() and d.name.isdigit())
 
-    all_rew_avg_all_trajs = []
-    all_rew_avg_top_trajs = []
-    all_infl_avg_all_trajs = []
-    all_infl_avg_top_trajs = []
     valid_iterations = []
+    trajs = defaultdict(list)
 
     for iteration in iterations:
         iteration_path = data_path / str(iteration)
         result = process_iteration_data(iteration_path, top_n)
 
         if result:
-            n_trajs, rew_avg_all_trajs, rew_avg_top_trajs, infl_avg_all_trajs, infl_avg_top_trajs = result
             valid_iterations.append(iteration)
-            all_rew_avg_all_trajs.append(rew_avg_all_trajs)
-            all_rew_avg_top_trajs.append(rew_avg_top_trajs)
-            all_infl_avg_all_trajs.append(infl_avg_all_trajs)
-            all_infl_avg_top_trajs.append(infl_avg_top_trajs)
+            for key in [
+                "top_n_trajs_dict",
+                "n_trajs",
+                "rew_avg_all_trajs",
+                "rew_avg_top_trajs",
+                "infl_avg_all_trajs",
+                "infl_avg_top_trajs",
+            ]:
+                trajs[key].append(result[key])
+
             if print_out:
                 print(f"\nIteration {iteration}:")
-                print(f"  Number of total entries: {n_trajs}")
-                print(f"  Reward average all trajectories: {rew_avg_all_trajs:.3f}")
+                print(f"  Number of total entries: {result['n_trajs']}")
+                print(f"  Reward average all trajectories: {result['rew_avg_all_trajs']:.3f}")
                 if top_n is not None and top_n > 0:
-                    print(f"  Reward average Top {top_n} Trajectories: {rew_avg_top_trajs:.3f}")
-                print(f"  Influence score average all trajectories: {infl_avg_all_trajs:.3f}")
+                    print(f"  Reward average Top {top_n} Trajectories: {result['rew_avg_top_trajs']:.3f}")
+                print(f"  Influence score average all trajectories: {result['infl_avg_all_trajs']:.3f}")
                 if top_n is not None and top_n > 0:
-                    print(f"  Influence score average Top {top_n} Trajectories: {infl_avg_top_trajs:.3f}")
+                    print(f"  Influence score average Top {top_n} Trajectories: {result['infl_avg_top_trajs']:.3f}")
 
         else:
             print(f"No valid data for iteration {iteration}")
 
     return (
         valid_iterations,
-        all_rew_avg_all_trajs,
-        all_rew_avg_top_trajs,
-        all_infl_avg_all_trajs,
-        all_infl_avg_top_trajs,
+        trajs["rew_avg_all_trajs"],
+        trajs["rew_avg_top_trajs"],
+        trajs["infl_avg_all_trajs"],
+        trajs["infl_avg_top_trajs"],
     )
```

Comment:
Not worth the time to change it rn, but maybe this could have also returned a dict?

Reply:
fixed